vendor/github.com/armon/go-metrics/prometheus/prometheus.go

```diff
@@ -39,19 +60,44 @@ type PrometheusSink struct {
 	expiration time.Duration
 }
 
-type PrometheusGauge struct {
+// GaugeDefinition can be provided to PrometheusOpts to declare a constant gauge that is not deleted on expiry.
+type GaugeDefinition struct {
+	Name []string
+	ConstLabels []metrics.Label
+	Help string
+}
+
+type gauge struct {
 	prometheus.Gauge
 	updatedAt time.Time
+	// canDelete is set if the metric is created during runtime so we know it's ephemeral and can delete it on expiry.
+	canDelete bool
+}
+
+// SummaryDefinition can be provided to PrometheusOpts to declare a constant summary that is not deleted on expiry.
+type SummaryDefinition struct {
+	Name []string
+	ConstLabels []metrics.Label
+	Help string
 }
 
-type PrometheusSummary struct {
+type summary struct {
 	prometheus.Summary
 	updatedAt time.Time
+	canDelete bool
+}
+
+// CounterDefinition can be provided to PrometheusOpts to declare a constant counter that is not deleted on expiry.
+type CounterDefinition struct {
```

Comment:
Does this need to `fmt`

Reply:
The Go fmt check did not bark at this file when I merged it upstream, and we're not able to change it here without a patch. I have it run as a post-write hook in my editor, so it should be good!